File: app/src/main/java/com/junerver/cloudnote/ui/activity/SplashActivity.java
package com.junerver.cloudnote.ui.activity;

import android.content.Intent;
import android.os.Bundle;
import android.os.Handler;
import android.support.v7.app.AppCompatActivity;
import android.widget.Toast;

import com.junerver.cloudnote.R;

import cn.bmob.v3.Bmob;
import cn.bmob.v3.BmobUser;
import cn.bmob.v3.exception.BmobException;
import cn.bmob.v3.listener.SaveListener;

public class SplashActivity extends AppCompatActivity {

    private static final String APPID = "0864d7f88908062c99fe51c58c65f254";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        // 初始化 Bmob SDK
        // 使用时请将第二个参数Application ID替换成你在Bmob服务器端创建的Application ID
        Bmob.initialize(this, APPID);
        setContentView(R.layout.activity_splash);

//		// 使用推送服务时的初始化操作
//		BmobInstallation.getCurrentInstallation(this).save();
//		// 启动推送服务
//		BmobPush.startWork(this, APPID);
        //一个延时跳转效果
        new Handler().postDelayed(new Runnable() {
            @Override
            public void run() {
                isFirstRun();
                SplashActivity.this.finish();
            }
        }, 1000);
    }


    public void isFirstRun() {

        //获取本地磁盘缓存的用户信息
        BmobUser bmobUser = BmobUser.getCurrentUser();
        if (bmobUser != null) {
            // 允许用户使用应用
            startActivity(new Intent(SplashActivity.this, MainActivity.class));
        } else {
            //缓存用户对象为空时， 可打开用户登录注册界面…
            startActivity(new Intent(SplashActivity.this, LoginActivity.class));
        }
    }


}

File: app/src/main/java/com/junerver/cloudnote/ui/activity/SplashActivity.kt
package com.junerver.cloudnote.ui.activity

import android.Manifest
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import com.junerver.cloudnote.R
import android.content.Intent
import android.os.Handler
import android.widget.Toast
import com.idealworkshops.idealschool.utils.SpUtils
import com.junerver.cloudnote.Constants
import com.junerver.cloudnote.ui.activity.MainActivity
import com.junerver.cloudnote.ui.activity.LoginActivity
import com.permissionx.guolindev.PermissionX

class SplashActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_splash)
        PermissionX.init(this)
            .permissions( Manifest.permission.READ_EXTERNAL_STORAGE)
            .explainReasonBeforeRequest()
            .request { allGranted, grantedList, deniedList ->
                if (allGranted) {
                    //一个延时跳转效果
                    Handler().postDelayed({
                        isFirstRun
                        finish()
                    }, 1000)
                } else {
                    Toast.makeText(this, "您拒绝了如下权限：$deniedList", Toast.LENGTH_SHORT).show()
                }
            }


    }//缓存用户对象为空时， 可打开用户登录注册界面…// 允许用户使用应用

    //获取本地磁盘缓存的用户信息
    val isFirstRun: Unit
        get() {

            //获取本地磁盘缓存的用户信息
            val bmobUser = SpUtils.decodeString(Constants.SP_USER_INFO)

            if (bmobUser.isNotEmpty()) {
                // 允许用户使用应用
                startActivity(Intent(this@SplashActivity, MainActivity::class.java))
            } else {
                //缓存用户对象为空时， 可打开用户登录注册界面…
                startActivity(Intent(this@SplashActivity, LoginActivity::class.java))
            }
        }
}